### PR TITLE
#164046257 Users can favorite and unfavorite articles

### PR DESCRIPTION
**Pull request by ElMonstro**
**Description**

This Pull request adds the ability for the user to favorite and unfavorite an article
<hr>
#### Type of change

- [x] new feature(non-breaking change which adds functionality).

<hr>

**How has this been tested**

- [x] End to End
- [x] Integration
<hr>

**How to test this manually**
- Clone the repository `https://github.com/andela/ah-legion-backend.git`
- run `git checkout --track origin/ft-favorite-unfavorite-articles-164046257`

After logging in and creating an article;
Test the following endpoints using postman:
- POST `/api/articles/<slug>/favorite/` to favorite an article
- GET `/api/articles/<slug>/favorite/` to test get favorite
- DELETE `/api/articles/<slug>/favorite/`to remove favorite
- GET  `/api/articles/favorites/` to test get favorites

**Checklist:**
- [x] This follows all the guidelines for this project.

#### Related Stories

[#164046255](https://www.pivotaltracker.com/n/projects/2313280/stories/#164046255)

### CODE REVIEW

**Review by MandelaK — APPROVED** [review]

**Review by Etomovich — APPROVED** [review]

**Review by BrianSerem — APPROVED** [review]

**Comment by abulojoshua1** [comment]
@ElMonstro, how are you returning an article's likes and dislikes total?

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -310,3 +311,91 @@ def get(self, request, slug):
             "dislikes": dislikes_count,
         }
         return Response(data=likes_dislikes, status=status.HTTP_200_OK)
+
+
+class FavoriteView(generics.CreateAPIView):
+    """This class creates a new like or dislike"""
```
Comment (abulojoshua1) [inline authors/apps/articles/views.py]: @ElMonstro, does the class create a like or dislike? I thought the PR was about favoriting.

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -310,3 +311,91 @@ def get(self, request, slug):
...
+    permission_classes = (IsAuthenticated,)
+    serializer_class = FavoriteSerializer
+
+    def post(self, request, slug):
+        """Creates a like"""
```
Comment (abulojoshua1) [inline authors/apps/articles/views.py]: @ElMonstro, does the class create a like or dislike? I thought the PR was about favoriting.

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -310,3 +311,91 @@ def get(self, request, slug):
...
+            article_id=article.id, user_id=request.user.pk).first()
+        if favorite:
+            message = {"detail": "Article removed from favorites"}
+            favorite.delete()
+            return Response(data=message, status=status.HTTP_204_NO_CONTENT)
+        not_found = {"detail": "Article not favorite"}
+        return Response(not_found, status.HTTP_404_NOT_FOUND)
+
+    def get(self, request, *args, **kwargs):
+        """"""
```
Comment (abulojoshua1) [inline authors/apps/articles/views.py]: @ElMonstro, why have empty doc strings?

File: authors/apps/articles/views.py [hunk 4]
```diff
@@ -310,3 +311,91 @@ def get(self, request, slug):
...
+            article_id=article.id, user_id=request.user.pk).first()
+        if favorite:
+            serializer = self.serializer_class(favorite)
+            return Response(data=serializer.data, status=status.HTTP_200_OK)
+        not_found = {"detail": "Article not favorite"}
+        return Response(not_found, status.HTTP_404_NOT_FOUND)
+
+
+class GetUserFavoritesView(APIView):
+    """Gets all articles' likes and dislikes"""
```
Comment (abulojoshua1) [inline authors/apps/articles/views.py]: @ElMonstro, does the class create a like or dislike? I thought the PR was about favoriting.

**Review by RachelleMaina — APPROVED** [review]

**Pull request merged**